### DISCUSSION TITLE
How to impose physical information in dde.nn.DeepONetCartesianProd.

### DISCUSSION THREAD

**Discussion by LanPeng-94** [lululxvi/deepxde#1874] [Q&A]
Suppose the u, y, and s data dimensions are (2000, 101), (200, 2), and (2000, 200), respectively. Using `dde.nn.DeepONetCartesianProd`, how do we impose automatic differentiation in the following code. Because the s_t and s dimensions do not match. 

```
def residual_net(self, u, x, t):
        s = self.net(u, x, t)
        
        s_t = torch.autograd.grad(s, t, grad_outputs=torch.ones_like(s), retain_graph=True, create_graph=True)[0]
        s_x = torch.autograd.grad(s, x, grad_outputs=torch.ones_like(s), retain_graph=True, create_graph=True)[0]
        s_xx = torch.autograd.grad(s_x, x, grad_outputs=torch.ones_like(s_x), retain_graph=True, create_graph=True)[0]
        res = s_t - 0.01 * s_xx - 0.01 * s**2 

        return res
```

**Comment by lululxvi** [comment]
https://deepxde.readthedocs.io/en/latest/demos/operator.html

**Reply by LanPeng-94** [reply]
Thank you for your reply.
I have studied your demo carefully.
In fact, I am curious about how you applied the automatic differentiation process to DeepONetCartesianRod.
Because it involves loop operations, do you use torch.vmap？
